c7n/utils.py

```diff
@@ -489,3 +489,18 @@ def format_string_values(obj, *args, **kwargs):
         return obj.format(*args, **kwargs)
     else:
         return obj
+
+def get_bad_snapshot(e):
```

Comment:
actually it was fine to just leave this on the class and use a `staticmethod` decorator to ease the testing, if we want to share it between multiple callers its probably best to move it to the ebs module, as its not really a general purpose utility its pretty specific to the ebs resource.

Reply:
makes sense-- I was thinking we would want to abstract out this type of string parsing to centralize changes if the messages themselves change the way that they're formatted, perhaps a more generic method like `get_bad_resource` to deal with this type of try/catch behavior

Reply:
yeah i think we'd considered something like that before, but there really isn't any good way to handle these generically, its error message parsing, and it varies on message content by api, and potentially also varies over time. if we can get this stuff out of the error handlers and into a set of functions that we can test thats a win.